src/autowiring/marshaller.h

```diff
@@ -128,34 +129,40 @@ namespace autowiring {
     }
   };
 
+#if defined(_MSC_VER) && _MSC_VER <= 1900 //This is reported resolved in VC++15
+#define AW_SNPRINTF sprintf_s
+#else
+#define AW_SNPRINTF snprintf
+#endif
+  //a version of the 32-bit NDK we use does not support to_string so we'll roll our own.
+  template<typename T, int bufferLen>
+  static inline std::string var_to_string(T value, const char* fmt) {
+    char buffer[bufferLen];
+    auto len = AW_SNPRINTF(buffer, sizeof(buffer), fmt, value);
+    return std::string(buffer, len);
+  }
+#undef AW_SNPRINTF
+
+  static inline std::string int_to_string(char c) { return var_to_string<char, 8>(c, "%hhd"); }
+  static inline std::string int_to_string(short c) { return var_to_string<short, 8>(c, "%hd"); }
+  static inline std::string int_to_string(int c) { return var_to_string<int, 16>(c, "%d"); }
+  static inline std::string int_to_string(long c) { return var_to_string<long, 32>(c, "%ld"); }
+  static inline std::string int_to_string(long long c) { return var_to_string<long long, 64>(c, "%lld"); }
```

Comment:
64? This template arg appears to be the size of the target buffer. For the minimum value of a `long long`, around negative 8 quintillion, I calculate that to be around 22 including the negative sign and null terminator, much less than 64.